napari/layers/_tests/test_layer_attributes.py

```diff
@@ -138,3 +140,40 @@ def test_get_value_3d_view_of_2d_image(ImageClass):
 def test_zero_scale_layer():
     with pytest.raises(ValueError, match='scale values of 0'):
         Image(np.zeros((64, 64)), scale=(0, 1))
+
+
+@pytest.mark.parametrize('Layer, data, ndim', layer_test_data)
+def test_sync_refresh_block(Layer, data, ndim):
+    my_layer = Layer(data)
+    my_layer.set_view_slice = MagicMock()
+
+    with my_layer._block_refresh():
+        my_layer.refresh()
+    my_layer.set_view_slice.assert_not_called
+
+    my_layer.refresh()
+    my_layer.set_view_slice.assert_called_once()
+
+
+@pytest.mark.parametrize('Layer, data, ndim', layer_test_data)
+def test_async_refresh_block(Layer, data, ndim):
+    from napari import settings
+
+    settings.get_settings().experimental.async_ = True
```

Comment:
Why is this line needed?

Reply:
@Czaki This test checks that [this codepath](https://github.com/napari/napari/blob/main/napari/layers/base/base.py#L1409) is also being handled correctly by the blocker and it's only executed when experimental async is on. We (@DragaDoncila) also did find [the async fixture](https://github.com/napari/napari/blob/main/napari/_qt/_tests/test_async_slicing.py#L27) and initially moved it to `conftest` but it requests `make_napari_viewer` and so it broke headless tests, so we just decided to set async manually here. Let me know if we should approach this a different way.

Reply:
But you put before this code. 

```
        if self._refresh_blocked:
            logger.debug('Layer.refresh blocked: %s', self)
            return
```

Reply:
@Czaki I'm not entirely sure what your objection is - can you please expand? We added that `if` statement to implement the blocking, but wanted to write the tests to ensure both the sync refresh and async refresh were blocked. 

If you're concerned about editing the settings directly, I think the `fresh_settings` [fixture](https://github.com/napari/napari/blob/main/napari/conftest.py#L195) will take care of that. Otherwise though we could move the test out to a separate file that can make use of the `enable_async` fixture.

Reply:
Ok. I see. This is a check for future refactor